google-cloud-spanner/src/test/java/com/google/cloud/spanner/InstanceAdminClientImplTest.java

```diff
@@ -128,9 +144,49 @@ public void createInstance() throws Exception {
   }
 
   @Test
-  public void getInstance() {
+  public void testCreateInstanceWithProcessingUnits() throws Exception {
+    OperationFuture<com.google.spanner.admin.instance.v1.Instance, CreateInstanceMetadata>
+        rawOperationFuture =
+            OperationFutureUtil.immediateOperationFuture(
+                "createInstance",
+                getInstanceProtoWithProcessingUnits(),
+                CreateInstanceMetadata.getDefaultInstance());
+    when(rpc.createInstance(
+            "projects/" + PROJECT_ID, INSTANCE_ID, getInstanceProtoWithProcessingUnits()))
+        .thenReturn(rawOperationFuture);
+    OperationFuture<Instance, CreateInstanceMetadata> operation =
+        client.createInstance(
+            InstanceInfo.newBuilder(InstanceId.of(PROJECT_ID, INSTANCE_ID))
+                .setInstanceConfigId(InstanceConfigId.of(PROJECT_ID, CONFIG_ID))
+                .setProcessingUnits(10)
+                .build());
+    assertTrue(operation.isDone());
+    assertEquals(INSTANCE_NAME, operation.get().getId().getName());
+  }
+
+  @Test
+  public void testCreateInstanceWithBothNodeCountAndProcessingUnits() throws Exception {
+    try {
+      client.createInstance(
+          InstanceInfo.newBuilder(InstanceId.of(PROJECT_ID, INSTANCE_ID))
+              .setInstanceConfigId(InstanceConfigId.of(PROJECT_ID, CONFIG_ID))
+              .setNodeCount(1)
+              .setProcessingUnits(100)
+              .build());
+      fail("missing expected exception");
+    } catch (IllegalArgumentException e) {
+      assertTrue(
+          e.getMessage()
+              .contains(InstanceAdminClientImpl.NOT_BOTH_NODE_COUNT_AND_PROCESSING_UNITS));
```

Comment:
in tests compare against the expected literal value, not a constant from the model code, so the test catches mistakes in constant definitions.

Reply:
Comparing against literal value now.